File: robocode2-server/src/main/java/net/robocode2/model/events/BulletHitBotEvent.java
package net.robocode2.model.events;

import lombok.Value;
import net.robocode2.model.Bullet;

/**
 * Event sent when a bullet hits a bot.
 * 
 * @author Flemming N. Larsen
 */
@Value
public class BulletHitBotEvent implements IEvent {

	/** Bullet that hit the bot */
	Bullet bullet;

	/** Bot id of the victim that was hit by the bullet */
	int victimId;

	/** Damage dealt to the victim */
	double damage;

	/** New energy level of the victim after damage */
	double energy;
}

File: robocode2-gui/src/main/kotlin/net/robocode2/gui/model/BulletHitBotEvent.kt
package net.robocode2.gui.model

data class BulletHitBotEvent(
        override val turnNumber: Int,
        val victimId: Int,
        val bullet: BulletState,
        val damage: Double,
        val energy: Double
) : Event(MessageType.BOT_HIT_BOT_EVENT.type, turnNumber)
